sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FileScanBuilder.scala

```diff
@@ -57,6 +64,27 @@ abstract class FileScanBuilder(
     StructType(fields)
   }
 
+  def pushFilters(partitionFilters: Seq[Expression], dataFilters: Seq[Expression]): Unit = {
+    this.partitionFilters = partitionFilters
+    this.dataFilters = dataFilters
+    this.pushedDataFilters = pushDataFilters(translateDataFilter)
+  }
+
+  protected def pushDataFilters(dataFilters: Array[Filter]): Array[Filter] = Array.empty[Filter]
```

Comment:
The two methods `pushFilters` and `pushDataFilters` are really confusing. Especially `pushFilters` also has `dataFilters` parameter.

Reply:
Or do you actually mean `pushDataFilters` -> `pushedDataFilters`?

Reply:
`pushFilters` means to push the filters (both `partitionFilters` and `dataFilters`) from spark to the file source. Since now file source has all the `dataFilters`, it can call the `pushDataFilters` method to find out which data filters can be processed in the file source, and set the `pushedDataFilters` to be these filters, so they later can be converted to data source filters and pushed down to the data sources. Hope this is clear :)